Card: Change the management of the inside `border-radius`

louismaximepiton opened the pull request:
Fixes #37010.

Couldn't spot any breaking change in the [example page](https://deploy-preview-37018--twbs-bootstrap.netlify.app/docs/5.3/components/card/), but I haven't dig deeper.

### Solution

Make the inner `border-radius` management universal inside the `.card`.
Leads to remove some unused classes like `.card-img-top` and `.card-img-bottom`.
May lighten the bundle size.

louismaximepiton committed:
Changing the management of card inner border-radius + update doc

mdo commented:
This removes two classes with the top and bottom variants. We'd need to deprecate those instead of remove outright—think you could tackle that?

louismaximepiton commented:
Hi mdo,

Not sure to completely understand what it is up to here. Should I mention it in the migration guide ? Should I use some deprecate comments ? Should I update the doc to tell that we don't use use those classes anymore ? It would be very helpful if you could light me up on this ! 

Thanks in advance !

mdo commented:
@louismaximepiton Sorry for the delay! We need to keep the original CSS around unmodified until v6. So let's add those `.card-img-top` and `.card-img-bottom` classes back in addition to the new one. Then we can add a deprecate note to the docs (I can handle this) and mention it in the migration guide (I can handle this too).

louismaximepiton committed:
Merge remote-tracking branch 'upstream/main' into main-lmp-card-inner-border-radius-fix

louismaximepiton committed:
Adding deprecation warning in scss + in docs

louismaximepiton commented:
No problem, I was busy on my side aswell. I tried something on the scss file, I'm not quite sure of the last commit on two parts:

- scss file: Reintroduced the classes but their behavior too, so people shouldn't notice the changes on normal use cases. I also added a deprecation warning message for those classes. Should I remove the behavior for `.card-img-top` and `.card-img-bottom`?
- docs file: Introduced a deprecation message here too based on the `.navbar-light` one.

May I let you the hand on the migration file ?

louismaximepiton committed:
Merge branch 'main' into main-lmp-card-inner-border-radius-fix

louismaximepiton committed:
Remove the changes in the doc

louismaximepiton committed:
Merge remote-tracking branch 'upstream/main' into main-lmp-card-inner-border-radius-fix

louismaximepiton committed:
Re arrange CSS

louismaximepiton committed:
Merge remote-tracking branch 'upstream/main' into main-lmp-card-inner-border-radius-fix
